ledger/stores/chunked_file_store.py

```diff
@@ -80,16 +93,25 @@ def getFileSize(file):
                 return chunk.numKeys
 
         path = os.path.join(dbDir, dbName)
-        os.mkdir(path)
+        if not os.path.isdir(path):
+            os.mkdir(path)
         if defaultFile:
             if self.chunkSize < getFileSize(defaultFile):
                 raise ValueError("Default file is larger than chunk size")
-            firstChunk = os.path.join(path, str(self.firstChunkIndex))
-            shutil.copy(defaultFile, firstChunk)
+            first_chunk = os.path.join(path, self._chunkIndexToFileName(self.firstChunkIndex))
+            if not os.path.exists(first_chunk.rsplit(os.path.sep,1)[0]):
+                os.mkdir(first_chunk.rsplit(os.path.sep,1)[0])
+            shutil.copy(defaultFile, first_chunk)
+
+    def _initDB(self, dbDir, dbName) -> None:
+        self.dbDir = dbDir
+        self.dbName = dbName
+        if not os.path.exists(self.dbDir):
+            os.makedirs(self.dbDir)
+        if not os.path.exists(os.path.join(dbDir, dbName)):
+            self._prepareFiles(dbDir, dbName, self._defaultFile)
 
-    def _initDB(self, dataDir, dbName) -> None:
-        super()._initDB(dataDir, dbName)
```

Comment:
It is slightly different in that it creates nested directories so we can't call super() anymore.